stacks/api/bucket/src/bucket-data.controller.ts

```diff
@@ -236,21 +255,33 @@ export class BucketDataController {
       req.res.header("Content-language", locale.best || locale.fallback);
     }
 
-    if (relation) {
-      const schema = await this.bs.findOne({_id: bucketId});
-      for (const propertyKey in schema.properties) {
-        const property = schema.properties[propertyKey];
-        if (property.type == "relation") {
-          aggregation.push(
-            ...buildRelationAggregation(
-              propertyKey,
-              property["bucketId"],
-              property["relationType"],
-              locale
-            )
-          );
+    if (typeof relation == "boolean") {
+      if (relation) {
+        const bucket = await this.bs.findOne({_id: bucketId});
+        for (const propertyKey in bucket.properties) {
+          const property = bucket.properties[propertyKey];
+          if (property.type == "relation") {
+            aggregation.push(
+              ...buildRelationAggregation(
+                propertyKey,
+                property["bucketId"],
+                property["relationType"],
+                locale
+              )
+            );
+          }
         }
       }
+    } else {
+      let fields: string[][] = [];
+      relation.forEach(pattern => fields.push(pattern.split(".")));
+
+      let buckets = await this.bs.find();
```

Comment:
If you mean fetching only related buckets, it will be only possible when creating aggregation recursively, fetching related schema for the current field. But it will cause wait for more promises to resolve. For example; 
Let's assume that you send a relation request to resolve like "a.b.c.d", it will fetch schema and create a lookup for b, then fetch schema and create a lookup for c, and the same steps for each relation chain.